hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcServer.java

```diff
@@ -1731,7 +1731,7 @@ protected List<RemoteLocation> getLocationsForPath(String path,
       final PathLocation location =
           this.subclusterResolver.getDestinationForPath(path);
       if (location == null) {
-        throw new IOException("Cannot find locations for " + path + " in " +
+        throw new NoLocationException("Cannot find locations for " + path + " in " +
```

Comment:
As we have a dedicated exception, we can pass the `path` and `class` parameters and build the message in `NoLocationException`